checkout-service/test/apiSpec.js

```diff
@@ -0,0 +1,74 @@
+const chai = require('chai'), chaiHttp = require('chai-http');
+const app = require('../server/index.js');
+
+chai.use(chaiHttp);
+chai.should();
+
+describe('API calls', () => {
+
+    let requester;
+
+    before(async () => {
+      requester = await chai.request(app).keepOpen();
+    });
+
+    describe('Single product API call - GET /priceandinventory/id/:productId', () => {
+      // valid ids are 1000-1099
+      const id = 1000;
+      const invalidId = 100;
+
+      it('should get a product\'s price and inventory count', (done) => {
+        requester.get(`/priceandinventory/id/${id}`)
+          .end((err, res) => {
+            res.should.have.status(200);
+            res.body.should.be.a('object');
+            done();
+          });
+      });
+
+      it('should respond with a 400 status code when requesting an invalid product id', (done) => {
+        requester.get(`/priceandinventory/id/${invalidId}`)
+          .end((err, res) => {
+            res.should.have.status(404);
+            done();
+          });
+      });
+    });
+
+    describe('Multiple products API call - POST /priceandinventory/id/multiple', () => {
+      // valid ids are 1000-1099
+      const ids = [1000, 1005, 1010, 1085];
+
+      it('should get multiple products\' prices and inventory counts', (done) => {
+        requester.post('/priceandinventory/id/multiple')
+          .send(ids)
+          .end((err, res) => {
+            res.should.have.status(200);
+            res.body.should.be.a('object');
```

Comment:
I like the unit test coverage that includes tests for both successful responses and errors. Since you are checking for status 200 and object on 2 different tests for this unit, my suggestion would be to broaden the test coverage within each of these tests to check for the existence of properties or keys. Maybe something like `res.body[0].productId.to.equal(...)`, or whatever syntax needed to check for properties.

From purely a reader's perspective, it would confirm that these two tests pass even if the response is an empty object!

Reply:
Done! This was some valuable insight as I had to refactor a couple of other files because I was unnecessarily sending the request as text which would've caused some issues later on. So thanks!